packages/react/src/primitives/StepperField/useStepper.ts

```diff
@@ -48,13 +48,11 @@ export const useStepper = ({
     getCorrectSteppingValue(min, max, step, defaultValue)
   );
 
-  // Same for controlled components on the first render because users could provide invalid intial value.
+  // Same for controlled components on the first render because users could provide invalid initial value.
   // It seems redundant afterwards but necessary for the first render
-  if (isControlled) {
-    controlledValue = getCorrectSteppingValue(min, max, step, controlledValue);
-  }
-
-  const value = isControlled ? controlledValue : uncontrolledValue;
+  const value = controlledValue
```

Comment:
same here. How we treat `null` value should be in line with React

Reply:
Weird, I initially changed this b/c TS didn't understand the isControlled boolean, but looks like it works now.

Reply:
Weird but it happens to me sometime. The type checking sometime lags behind the updates and may need to reload the compiler